### ISSUE TITLE
algolia_reindex command resets the ranking formula.

### ISSUE THREAD

**Issue by PLNech** [algolia/algoliasearch-django#238]
From [support](https://secure.helpscout.net/conversation/503753260/47217/?folderId=1106372):

>We use Algolia with Django and once a day we reindex our data. We need to reindex it even with enabled auto-indexing. Unfortunately, algolia_reindex command resets our ranking formula.

**Comment by noclat** [comment]
What's the status on this one? A customer just pointed out again the ranking reset bug. Shouldn't it be fixed?

**Comment by PLNech** [comment]
Normally fixed since [1.5.4](https://github.com/algolia/algoliasearch-django/releases/tag/1.5.4), can you check the customer's current version?

**Comment by noclat** [comment]
They're on 1.5.4, and they still experience the issue. So I keep that issue open and let's reach back [#517675705](https://secure.helpscout.net/conversation/517675705?folderId=1106372) when fixed!

**Comment by abakar** [comment]
Hey guys. I actually told about that on January 16 and have got no answer untill today:
<img width="1243" alt="screenshot" src="https://user-images.githubusercontent.com/1752096/35817995-eff0f0ee-0aaf-11e8-88d9-7964ca7ea6cd.png">

**Comment by PLNech** [comment]
Hi @abakar, it seems your follow-up email slipped through last week. Sincere apologies for that.

I spent the day investigating further, see the open PR #241 which adds two tests that seem to confirm expected behavior:
- If you define settings in your [registered `AlgoliaIndex`](https://github.com/algolia/algoliasearch-django#quick-start), those are applied when you reindex
- If you don't set specific settings when registering your Model, your current settings are kept over reindex.

Can you check if your app defines settings by registering a custom `AlgoliaIndex` subclass? 
- If it does, this explains the settings being reset: defining `settings` in your `AlgoliaIndex` subclass is a way to get these settings applied when reindexing (which indeed means you would lose any other setting in that index)
- If it doesn't, could you share here the code where you call `algoliasearch.register` and the definition of any mentioned Model / AlgoliaIndex? This would help pinpoint the difference between our succeeding tests and your failing use-case.

**Comment by Haizzz** [comment]
We just started using this library and everytime we reindex our ranking formula is still kept but the synonyms and query rules get wiped. We created a hotfix specific to our use case that export the query rules and synonyms, reindex then import them again. The code is below. Hope this helps
```python
from algoliasearch_django.management.commands.algolia_reindex import Command as AlgoliaCommand
from algoliasearch import algoliasearch
from shops.index import ProductIndex
from django.conf import settings


class Command(AlgoliaCommand):
    help = 'Reindex all models to Algolia with Feedmee fixes'

    def handle(self, *args, **options):
        # FEEDMEE fix, export then import synonyms and queryrules
        client = algoliasearch.Client(settings.ALGOLIA["APPLICATION_ID"],
                                      settings.ALGOLIA["API_KEY"])
        index = client.init_index(settings.ALGOLIA["NON_REPLICA"])
        rules = []
        synonyms = []
        for r in index.iter_rules():
            rules.append(r)
        for s in index.iter_synonyms():
            synonyms.append(s)
        print("Found rules: {}".format(rules))
        print("Found synonyms: {}".format(synonyms))
        # the normal command
        super(Command, self).handle(*args, **options)
        # save them
        response = index.batch_rules(rules, forward_to_replicas=True)
        index.wait_task(response['taskID'])
        print("Saved rules with response: {}".format(response))
        response = index.batch_synonyms(synonyms, forward_to_replicas=True)
        index.wait_task(response['taskID'])
        print("Saved synonyms with response: {}".format(response))
```

**Comment by abakar** [comment]
Hi @PLNech,

You are right — I have index settings with "searchableAttributes" and "attributesForFaceting" keys. I've checked now "algolia_reindex" command without any settings and it didn't erase ranking formula.

Thank you!

**Comment by PLNech** [comment]
Hi @Haizzz, thanks for your report! Indeed, rules and synonyms were not handled.
I addressed this in b14bc87, we'll release an update today 🙂

**Comment by PLNech** [comment]
Fixed in release [1.5.5](https://pypi.python.org/pypi/algoliasearch-django/1.5.5).
